python-clusters/create-aks-cluster/cluster.json

```
@@ -155,6 +155,36 @@
             "mandatory": false,
             "defaultValue": "Standard"
         },
+        {
+            "name": "upgradeChannel",
+            "label": "Cluster upgrades",
+            "description": "Determines whether and when the cluster is automatically upgraded based on new K8S or AKS releases",
+            "type": "SELECT",
+            "selectChoices": [
+                {"value":"none", "label":"No automatic upgrade"},
+                {"value":"patch", "label":"Latest patch version on currently used minor"},
+                {"value":"stable", "label":"Latest patch release on next to latest minor"},
+                {"value":"rapid", "label":"Latest patch release on latest minor"},
+                {"value":"node-image", "label":"Latest node image version"}
+            ],
+            "mandatory": true,
+            "defaultValue": "none"
+        },
+        {
+            "name": "nodeOSUpgradeChannel",
+            "label": "Node OS upgrades",
+            "description": "Determines whether and when the OS in the cluster nodes is upgraded",
+            "type": "SELECT",
+            "selectChoices": [
+                {"value":"None", "label":"No automatic upgrade"},
+                {"value":"Unmanaged", "label":"Upgrades handled by the OS"},
+                {"value":"SecurityPatch", "label":"Weekly upgrades handled by AKS - security patches only"},
```

```suggestion
                {"value":"SecurityPatch", "label":"Weekly upgrades handled by AKS - security patches only (preview)"},
```
Maybe we could had a preview flag to let them know they need the FF activated. The error is explicit but it's unclear the ff only impact this channel.

---

agree. Not much more we can do anyway, I don't want to go all the way and probe which preview features are active (you'll need elevated permissions for that anyway, I bet)